diff --git a/mrbgems/mruby-time/src/time.c b/mrbgems/mruby-time/src/time.c
index 59dcd80be..55442412d 100644
--- a/mrbgems/mruby-time/src/time.c
+++ b/mrbgems/mruby-time/src/time.c
@@ -84,7 +84,7 @@ double round(double x) {
 /** end of Time class configuration */

 /* protection against incorrectly defined _POSIX_TIMERS */
-#if defined(_POSIX_TIMERS) && (_POSIX_TIMERS + 0) > 0 && defined(CLOCK_REALTIME)
+#if defined(_POSIX_TIMERS) && (_POSIX_TIMERS + 0) > 0 && defined(CLOCK_REALTIME) && !defined(DISABLE_CLOCK_GETTIME)
 # define USE_CLOCK_GETTIME
 #endif
